src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -49,10 +48,17 @@ public PreferencesView(DynamoView dynamoView)
             {
                 viewModel = viewModelTemp;
             }
-            
+
+            SyncPreferenceSettingsAndViewModels();
             InitRadioButtonsDescription();
         }
 
+        // solution 1 - preferencesView is reconstructed each time the prefs window is opened
+        // update any state we need to here on any child viewModels.
+        private void SyncPreferenceSettingsAndViewModels()
+        {
+            viewModel?.PackagePathsViewModel?.InitializeRootLocations();
```

Comment:
I used a slightly different way in my PR: `PreferencesView::SetupPreferencesViewModel()` is called everytime the dialog is opened.

Reply:
@sm6srw - yeah that existing method could work as well, I can just call initialize from there - I'm assuming we're eventually going to find other preferences (besides package prefs) that will now be out of sync with the preferences dialog.

It's not usual for these to be set from anywhere but the UI - EXCEPT, extensions probably set them all over the place - so I think it's good to have a central location to sync them. I'm okay with moving over to this method if you agree?

Reply:
Yes, I do. Lets reuse the existing method.